### PR TITLE
Add docs for csi-driver-spiffe v0.6.0

### PR DESCRIPTION
**Pull request by SgtCoDFish**
Also adds security documentation and other improvements for csi-driver-spiffe docs in general

### CODE REVIEW

File: content/docs/usage/csi-driver-spiffe/README.md
Comment (maelvls) [inline content/docs/usage/csi-driver-spiffe/README.md:162]: Is it explained anywhere why `app.approver.signerName` isn't meant to be used along with  `app.runtimeIssuanceConfigMap`? Or why this field isn't needed anymore?

Comment (SgtCoDFish) [reply content/docs/usage/csi-driver-spiffe/README.md:162]: See "Simpler Install with no signerName" [on the release notes](https://github.com/cert-manager/csi-driver-spiffe/releases/tag/v0.6.0)

I considered documenting this a bit but I think the signerName setting is either an antipattern or else so specifically an extreme power user feature that it wasn't really worth it.

Comment (maelvls) [reply content/docs/usage/csi-driver-spiffe/README.md:162]: Thanks, the link you shared was super useful. You are right, let's just recommend people to not use this flag by not showing it at all in the installation instructions 👍

File: content/docs/usage/csi-driver-spiffe/README.md
Comment (maelvls) [inline content/docs/usage/csi-driver-spiffe/README.md:118]: Thanks, this section is super useful; I haven't found a design doc for csi-driver-spiffe, but what you wrote makes a good description of the design!

**Review by maelvls — APPROVED** [review]
I like the fact that you wrote a new paragraph talking about the design philosophy and limitations of csi-driver-spiffe "borrowing the token of pods".

/lgtm
/approve
/hold in case there is any more comment

**Comment by SgtCoDFish** [comment]
Gonna unhold since v0.6.0 is already out. Thanks for the review! 😁 

/unhold

**Pull request merged**
